modules/cudf/src/series.ts

```diff
@@ -648,6 +648,65 @@ export class AbstractSeries<T extends DataType = any> {
     return this.__construct(this._col.gather(selection._col));
   }
 
+  /**
+   * perform nLargest or nSmallest operation
+   * @ignore
+   */
+  _nLargestOrSmallest(ascending: boolean, n: number, keep: DuplicateKeepOption) {
+    if (keep == DuplicateKeepOption.first) {
+      return this.sortValues(!ascending).head(n < 0 ? 0 : n);
+    } else if (keep == DuplicateKeepOption.last) {
+      return n <= 0 ? Series.new({type: this.type, data: new Array(0)})
+                    : this.sortValues(ascending).tail(n).reverse();
+    } else {
+      throw new TypeError('keep must be either "first" or "last"');
+    }
+  }
```

Comment:
Why not have these logic branches in each of the functions directly?

Reply:
Do you mean...

```ts
nLargest(n = 5, keep: DuplicateKeepOption = DuplicateKeepOption.first) {
    if (keep == DuplicateKeepOption.first) {
      return this.sortValues(!ascending).head(n < 0 ? 0 : n);
    } else if (keep == DuplicateKeepOption.last) {
      return n <= 0 ? Series.new({type: this.type, data: new Array(0)})
                    : this.sortValues(ascending).tail(n).reverse();
    } else {
      throw new TypeError('keep must be either "first" or "last"');
    }
}

nSmallest(n = 5, keep: DuplicateKeepOption = DuplicateKeepOption.first) {
    if (keep == DuplicateKeepOption.first) {
      return this.sortValues(!ascending).head(n < 0 ? 0 : n);
    } else if (keep == DuplicateKeepOption.last) {
      return n <= 0 ? Series.new({type: this.type, data: new Array(0)})
                    : this.sortValues(ascending).tail(n).reverse();
    } else {
      throw new TypeError('keep must be either "first" or "last"');
    }
}

```
(ignoring any syntax errors / incorrect variables).

I basically set it up as it is in `CUDF` https://github.com/rapidsai/cudf/blob/90fe10b2947c9caee998f6236d432f3742c3504e/python/cudf/cudf/core/series.py#L3242

I don't really have a preference, I'm fine to change it.

Reply:
Oh my bad, I didn't realize the switching on `keep` wasn't about the largest/smallest, but the first/last.